modules/code-deployment/src/main/java/org/apache/ignite/internal/deployunit/DeploymentManagerImpl.java

```diff
@@ -0,0 +1,424 @@
+                        }
+                    }
+                });
+        return result;
+    }
+
+    @Override
+    public void start() {
+        unitsFolder = unitsFolder.resolve(configuration.deploymentLocation().value());
+        clusterService.messagingService().addMessageHandler(DeployUnitMessageTypes.class,
+                (message, senderConsistentId, correlationId) -> {
+                    if (message instanceof DeployUnitRequest) {
+                        processDeployRequest((DeployUnitRequest) message, senderConsistentId, correlationId);
+                    } else if (message instanceof UndeployUnitRequest) {
+                        processUndeployRequest((UndeployUnitRequest) message, senderConsistentId, correlationId);
+                    }
+                });
+    }
+
+    private void processDeployRequest(DeployUnitRequest executeRequest, String senderConsistentId, long correlationId) {
+        doDeploy(executeRequest).whenComplete((success, throwable) -> {
+            DeployUnitResponseBuilder builder = DeployUnitResponseImpl.builder();
+            if (throwable != null) {
+                builder.error(throwable);
+            }
+            clusterService.messagingService().respond(senderConsistentId,
+                    builder.build(), correlationId);
+        });
+    }
+
+    private void processUndeployRequest(UndeployUnitRequest executeRequest, String senderConsistentId, long correlationId) {
+        try {
+            Path unitPath = unitsFolder
+                    .resolve(executeRequest.id())
+                    .resolve(executeRequest.version());
+
+            IgniteUtils.deleteIfExistsThrowable(unitPath);
+        } catch (IOException e) {
+            LOG.error("Failed to undeploy unit " + executeRequest.id() + ":" + executeRequest.version(), e);
+            clusterService.messagingService()
+                    .respond(senderConsistentId, UndeployUnitResponseImpl.builder().error(e).build(), correlationId);
+            return;
+        }
+
+        clusterService.messagingService()
+                .respond(senderConsistentId, UndeployUnitResponseImpl.builder().build(), correlationId);
+    }
+
+    private CompletableFuture<Boolean> doDeploy(DeployUnitRequest executeRequest) {
+        String id = executeRequest.id();
+        String version = executeRequest.version();
+        try {
+            Path unitPath = unitsFolder
+                    .resolve(executeRequest.id())
+                    .resolve(executeRequest.version())
+                    .resolve(executeRequest.unitName());
+            Path unitPathTmp = unitPath.resolveSibling(unitPath.getFileName() + ".tmp");
+            Files.createDirectories(unitPathTmp.getParent());
+            Files.write(unitPathTmp, executeRequest.unitContent(), CREATE, SYNC, TRUNCATE_EXISTING);
+            Files.move(unitPathTmp, unitPath, ATOMIC_MOVE, REPLACE_EXISTING);
```

Comment:
Not enough empty lines in your code, so it's pretty hard to read. Usually the code in Ignite is more sparse, I guess it must be mentioned somewhere in code style guidelines